src/scripts/main.js

```diff
@@ -1,3 +1,92 @@
-'use strict';
+document.addEventListener('DOMContentLoaded', function () {
+  const table = document.querySelector('.field');
+  const appendRowBtn = document.querySelector('.append-row');
+  const removeRowBtn = document.querySelector('.remove-row');
+  const appendColumnBtn = document.querySelector('.append-column');
+  const removeColumnBtn = document.querySelector('.remove-column');
 
-// write code here
+  const maxRows = 10;
+  const minRows = 2;
+  const maxColumns = 10;
+  const minColumns = 2;
+
+  updateButtonStates();
+
+  appendRowBtn.addEventListener('click', appendRow);
+  removeRowBtn.addEventListener('click', removeRow);
+  appendColumnBtn.addEventListener('click', appendColumn);
+  removeColumnBtn.addEventListener('click', removeColumn);
+
+  function appendRow() {
+    if (table.rows.length < maxRows) {
+      const newRow = table.insertRow();
+
+      for (let i = 0; i < table.rows[0].cells.length; i++) {
+        const newCell = newRow.insertCell();
+
+        newCell.textContent = '';
+      }
+
+      updateButtonStates();
+    }
+  }
+
+  function removeRow() {
+    if (table.rows.length > minRows) {
+      table.deleteRow(-1);
+
+      updateButtonStates();
+    }
+  }
+
+  function appendColumn() {
+    if (table.rows[0].cells.length < maxColumns) {
+      for (let i = 0; i < table.rows.length; i++) {
+        const newCell = table.rows[i].insertCell();
+
+        newCell.textContent = '';
+      }
+
+      updateButtonStates();
+    }
+  }
+
+  function removeColumn() {
+    if (table.rows[0].cells.length > minColumns) {
```

Comment:
Similar to the appendColumn function, this line assumes that the table will always have at least one row. If the table is initially empty, accessing `table.rows[0].cells.length` will result in an error. Consider adding a check to ensure the table has rows before accessing `table.rows[0].cells.length`.